source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,351 @@
+Minecraft's objects. Whenever any data is requested from or offered to ``DataHolders`` that exist in Vanilla
+Minecraft, those calls end up being handled by a ``DataProcessor`` or a ``ValueProcessor``.
+
+For your name, you should use the name of the ``DataManipulator`` interface and append ``Processor``. Thus for ``HealthData`` we create a ``HealthDataProcessor``.
+
+In order to reduce boilerplate code, the ``DataProcessor`` should inherit from the appropriate abstract class in
+the ``org.spongepowered.common.data.processor.common`` package. Since health can only be present on certain
+entities, we can make use of the ``AbstractEntityDataProcessor`` which is specifically
+targeted at ``Entities`` based on ``net.minecraft.entity.entity``.
+
+.. code-block:: java
+
+    public class HealthDataProcessor extends AbstractEntityDataProcessor<EntityLivingBase, HealthData, ImmutableHealthData> {
+        public HealthDataProcessor() {
+            super(EntityLivingBase.class);
+        }
+        [...]
+    }
+
+Depending on which abstraction you use, the methods you have to implement may differ greatly, depending on how
+much implementation work already could be done in the abstract class. Generally, the methods can be categorized.
+
+Validation Methods
+~~~~~~~~~~~~~~~~~~
+
+Always return a boolean value. If the method is called ``supports()`` it should perform a general check if the supplied target generally supports the kind of data handled by our ``DataProcessor``.
+
+For our ``HealthDataProcessor`` ``supports()`` is implemented by the ``AbstractEntityDataProcessor``. Per
+default, it will return true if the supplied argument is an instance of the class specified when calling the
+``super()`` constructor.
+
+Instead, we are required to provide a ``doesDataExist()`` method. Since the abstraction does not know how to
+obtain the data, it leaves this function to be implemented. As the name says, the method should check if the data
+already exists on the supported target. For the ``HealthDataProcessor``, this always returns true, since every
+living entity always has health.
+
+.. code-block:: java
+
+    protected boolean doesDataExist(EntityLivingBase entity) {
+        return true;
+    }
+
+Setter Methods
+~~~~~~~~~~~~~~
+
+A setter method receives a ``DataHolder`` of some sort and some data that should be applied to it, if possible.
+
+The ``DataProcessor`` interface defines a ``set()`` method accepting a ``DataHolder`` and a ``DataManipulator`` which returns a ``DataTransactionResult``. Depending on the abstraction class used, some of the necessary functionality might already be implemented.
+
+In this case, the ``AbstractEntityDataProcessor`` takes care of most of it and just requires a method to set some values to return ``true`` if it was successful and ``false`` if it was not. Creation of the ``DataTransactionResult`` is fully handled by the abstract class.
+
+.. code-block:: java
```

Comment:
You should likely comment on the map being passed in is just the values the `DataManipulator` you're using is being mapped of `Key` to the underlying value.